From 146bb34d0e04e5174b7e6973ec0a93bf38548647 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 22 Feb 2024 09:10:45 +0100
Subject: [PATCH 0647/1695] dts: stm32: add HSI/CSI calibration on
 STM32MP135F-DK

Default enables HSI/CSI calibration on STM32MP135F-DK boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3cd7094aacb389af2a483b32694b30396a3b50ae dts: stm32: add HSI/CSI calibration on STM32MP135F-DK]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I8325b71fb1221d6bd3100753e3e04e483d645cfc
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/286373
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362466
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 71e918162..e3746b261 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -368,6 +368,16 @@
 	st,non-secure-otp-provisioning;
 };
 
+&osc_calibration {
+	csi-calibration {
+		status = "okay";
+	};
+
+	hsi-calibration {
+		status = "okay";
+	};
+};
+
 &pwr_irq {
 	status = "okay";
 	wakeup-gpios = <&gpiof 8 GPIO_ACTIVE_HIGH>, <&gpioi 3 GPIO_ACTIVE_HIGH>, <0>, <0>, <&gpioi 2 GPIO_PULL_UP>, <0>;
@@ -622,6 +632,14 @@
 	};
 };
 
+&timers12 {
+	status = "okay";
+
+	counter {
+		status = "okay";
+	};
+};
+
 &tzc400 {
 	memory-region = <&optee_framebuffer>;
 };
-- 
2.47.2

